src/Components/Location.js

```diff
@@ -0,0 +1,89 @@
+import React, { useState, useEffect } from "react";
+
+export default function Location() {
+    const [location, setLocation] = useState([]);
+    const [nameButton, setNameButton] = useState('');
+    const [climateButton, setClimateButton] = useState('');
+    const [terrainButton, setTerrainButton] = useState('')
+    const [name, setName] = useState('');
+    const [climate, setClimate] = useState('');
+    const [terrain, setTerrain] = useState('');
+
+    useEffect(() => {
+        fetch(`https://ghibliapi.herokuapp.com/locations`)
+            .then((response) => response.json())
+            .then((data) => setLocation(data))
+    })
+
```

Comment:
1.  capture the 'key' that we are going to sort by - this should come from the button
2. Pass that key to a sort function - google js.sort() 
3. compare (a,b) => a[key] <b[key] return 1
4. take the return from the sort and pass that to setLocation()